src/query/query_executor.c

```diff
@@ -2248,6 +2249,10 @@ qexec_clear_xasl (THREAD_ENTRY * thread_p, xasl_node * xasl, bool is_final)
       pg_cnt += qexec_clear_agg_orderby_const_list (thread_p, xasl, is_final);
     }
 
+  if (xasl->sq_cache_flag != 0)
```

Comment:
Why should we check if `sq_cache_flag` is 0 instead of checking if `sq_cache_ht` is NULL?

```c++
if (xasl->sq_cache_flag != 0)
  {
    assert (xasl->sq_cache_ht != NULL);
  }
```